evmrpc/testutils.go

```diff
@@ -162,4 +170,56 @@ func init() {
 		panic(err)
 	}
 	time.Sleep(1 * time.Second)
+
+	to := common.HexToAddress("010203")
+	txData := ethtypes.DynamicFeeTx{
+		Nonce:     1,
+		GasFeeCap: big.NewInt(10),
+		Gas:       1000,
+		To:        &to,
+		Value:     big.NewInt(1000),
+		Data:      []byte("abc"),
+		ChainID:   big.NewInt(1),
+	}
+	mnemonic := "fish mention unlock february marble dove vintage sand hub ordinary fade found inject room embark supply fabric improve spike stem give current similar glimpse"
+	derivedPriv, _ := hd.Secp256k1.Derive()(mnemonic, "", "")
+	privKey := hd.Secp256k1.Generate()(derivedPriv)
+	testPrivHex := hex.EncodeToString(privKey.Bytes())
+	key, _ := crypto.HexToECDSA(testPrivHex)
+	evmParams := EVMKeeper.GetParams(Ctx)
+	ethCfg := evmParams.GetChainConfig().EthereumConfig(big.NewInt(1))
+	signer := ethtypes.MakeSigner(ethCfg, big.NewInt(Ctx.BlockHeight()), uint64(Ctx.BlockTime().Unix()))
+	tx := ethtypes.NewTx(&txData)
+	tx, err = ethtypes.SignTx(tx, signer, key)
+	if err != nil {
+		panic(err)
+	}
+	typedTx, err := ethtx.NewDynamicFeeTx(tx)
+	if err != nil {
+		panic(err)
+	}
+	msg, err := types.NewMsgEVMTransaction(typedTx)
+	if err != nil {
+		panic(err)
+	}
+	b := TxConfig.NewTxBuilder()
+	if err := b.SetMsgs(msg); err != nil {
+		panic(err)
+	}
+	Tx = b.GetTx()
+	if err := EVMKeeper.SetReceipt(Ctx, tx.Hash(), &types.Receipt{
+		From:              "0x123456789012345678902345678901234567890",
```

Comment:
How come this starts with `0x123...` but in your tests you're checking for `0x0123...`?

Reply:
i think it padded a zero at the beginning before `0x123456789012345678902345678901234567890` has odd length. Changed it to be even length to make it less confusing